lib/JobProcessor.js

```diff
@@ -276,12 +277,25 @@ JobProcessor.prototype.jobProcessorAsync = function() {
             return;
         }
         // generate tile and repeat
-        return self.processOneTileAsync(iterValue).then(() => {
-            if (self.isShuttingDown) {
-                throw new Err('Shutting down');
-            }
-            return self.jobProcessorAsync();
-        });
+        // if timeout is 0 or not specified, e.g. null, keep old behavior
+        if (!this.tileTimeOut){
+            return self.processOneTileAsync(iterValue).then(() => {
+                if (self.isShuttingDown) {
+                    throw new Err('Shutting down');
+                }
+                return self.jobProcessorAsync();
+            });
+        } else {
+            // set timeout for tile process in case mapnik get stucked
+            return self.processOneTileAsync(iterValue).timeout(this.tileTimeOut).then(() => {
```

Comment:
Rather than doing an if/else conditional, I think it would be cleaner to initialize `this.tileTimeOut` to `tileTimeOut || 0` above, and then run the code currently in the else block in all cases.  (Note that [`.timeout`](http://bluebirdjs.com/docs/api/timeout.html) is fundamentally just [promise-wrapping](https://github.com/petkaantonov/bluebird/blob/49da1ac256c7ee0fb1e07679791399f24648b933/src/timers.js#L70-L91) [`setTimeout()`](https://nodejs.org/api/timers.html#timers_settimeout_callback_delay_args).)

Reply:
Done.